File: solutions/0108_Convert_Sorted_Array_to_Binary_Search_Tree/recur.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return toBST(nums, 0, nums.length - 1);
    }

    private TreeNode toBST(int[] nums, int s, int t) {
        if (s > t) {
            return null;
        }
        int mid = (s + t) / 2;
        TreeNode root = new TreeNode(nums[mid]);
        root.left = toBST(nums, s, mid - 1);
        root.right = toBST(nums, mid + 1, t);
        return root;
    }
}

File: solutions/0226_Invert_Binary_Tree/recur.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        self.recur(root)
        return root

    def recur(self, node):
        if not node:
            return
        self.recur(node.left)
        self.recur(node.right)
        node.left, node.right = node.right, node.left
